pkg/controller/pod_volume_restore_controller.go

```diff
@@ -151,30 +157,36 @@ func UpdatePVRStatusToFailed(c client.Client, ctx context.Context, pvr *velerov1
 	return c.Patch(ctx, pvr, client.MergeFrom(original))
 }
 
-func (c *PodVolumeRestoreReconciler) shouldProcess(ctx context.Context, log logrus.FieldLogger, pvr *velerov1api.PodVolumeRestore) (bool, *corev1api.Pod, error) {
-	if !isPVRNew(pvr) {
-		log.Debug("PodVolumeRestore is not new, skip")
-		return false, nil, nil
-	}
-
+func (c *PodVolumeRestoreReconciler) shouldProcess(ctx context.Context, log logrus.FieldLogger, pvr *velerov1api.PodVolumeRestore) (bool, bool, *corev1api.Pod, error) {
+	// We need to check pods status
 	// we filter the pods during the initialization of cache, if we can get a pod here, the pod must be in the same node with the controller
 	// so we don't need to compare the node anymore
 	pod := &corev1api.Pod{}
 	if err := c.Get(ctx, types.NamespacedName{Namespace: pvr.Spec.Pod.Namespace, Name: pvr.Spec.Pod.Name}, pod); err != nil {
 		if apierrors.IsNotFound(err) {
 			log.WithError(err).Debug("Pod not found on this node, skip")
-			return false, nil, nil
+			return false, false, nil, nil
```

Comment:
This will result in a false failing.
See the node-agent server code [here](https://github.com/vmware-tanzu/velero/blob/main/pkg/cmd/cli/nodeagent/server.go#:~:text=on%20this%20node.-,cacheOption,%7B,-SelectorsByObject), the client caches the pod that is scheduled in the current node only.
That means only one controller (in the node that the pod is scheduled) could get the restored pod, all other controllers get errorNotFound here.
If the code is changed like this, the PVRs will be marked as failed inevitably.

Reply:
If pod is  not found,  it will not mark pvr failed. 
This commit mainly covers the updated scenarios after the deployment backup.
We use the old version to restore the deployment and pod. The old pod will be immediately deleted by the rs controller, which will cause the entire restore to be unable to continue. At this time, pvr will remain in InProgress until timeout or a pod with the same name is created.
I think we can consider this scenario。
Of course  I only tested it on 1.9， PR #5760  I didn't test this scenario, but I don't think it's very good。

Reply:
Yes, you are right, the PVR is marked as failed only when the pod is being deleted. Meanwhile, #5760 still covers the same pod deletion scenarios. Therefore, unless it is a scenario that #5760 doesn't cover, we don't want to involve duplicated logics, noticeably, #5760 fix several similar problems, not only the current one.

Reply:
For the case:

> We use the old version to restore the deployment and pod. The old pod will be immediately deleted by the rs controller, which will cause the entire restore to be unable to continue.

Could you explain more? As the Velero [restore policy](https://velero.io/docs/v1.10/restore-reference/#restore-existing-resource-policy:~:text=ns%2D1.-,Restore%20existing%20resource%20policy,-By%20default%2C%20Velero), Velero by default doesn't support to overwrite existing objects. Do you mean in the case you mentioned, something has escaped and broken this policy?

Reply:
Sorry，you are right, my code will mark the completed pvr as failed. 
When I have time to upgrade my velero, go to test PR #5760 .